src/Components/Endpoints/src/Builder/RazorComponentEndpointFactory.cs

```diff
@@ -38,6 +40,7 @@ internal class RazorComponentEndpointFactory
         builder.Metadata.Add(new SuppressLinkGenerationMetadata());
         builder.Metadata.Add(HttpMethodsMetadata);
         builder.Metadata.Add(new ComponentTypeMetadata(pageDefinition.Type));
+        builder.Metadata.Add(new RootComponentMetadata(rootComponent));
```

Comment:
Who's going to use the `RootComponentMetadata`? It doesn't look like anything references it currently.

I see that the `rootComponent` `Type` now flows to `RazorComponentEndpointInvoker` via a lambda capture below. Does that mean `RootComponentMetadata` can be deleted?

If this is meant for public consumption or some future code we're going to write, we should at least write a test for now.

Reply:
Root component metadata is for public consumption so that the endpoint can be identified.

Reply:
Done